Fix compatibility with GitHub Enterprise; More logging; Increase retries

GUI opened the pull request:
This makes 3 small changes that are semi-related, but I'm happy to split these up or remove any of these pieces if you'd prefer not to include them. The main thing I was trying to fix was compatibility with GitHub Enterprise, but in the process I tackled a couple of other partially related things it led to.

1. Fix compatibility when running on GitHub Enterprise. In GitHub Enterprise, the GitHub Actions app ID may not be the hard-coded `15368` value that was being referenced here (it could be some other random ID), so this meant the code was never identifying its own check. Without knowing its own check, the checks would loop indefinitely until the retry limit was hit, since the `allcheckspassed` check would be waiting for itself.

   To fix this, I shifted the "own" logic to use the app `slug`, which I believe will always be `github-actions` in both public GitHub.com and private GitHub Enterprise installs.

2. In the process of debugging this I added some more verbose logging that I thought might be more generally useful to diagnose any issues like this that might arise:

    - Log a warning if the `ownCheck` can't be determined, since I believe this indicates an issue that will then cause the polling to continue indefinitely until `retries` is hit.
    - Log a warning if the job name for the "own" check can't be determined. This might happen if you miss the required `contents: read` permissions noted in the README and could similarly lead to indefinite looping.
    - Log extra info lines to show the name of the check being waited on in each loop (this is purely informational, but I thought it might help understanding what's going on).

3. Removes the `360` maximum value (enforced by `validateIntervalValues`) from the `retries` parsing. In light of intervals now possibly being fractions (https://github.com/wechuli/allcheckspassed/pull/6), the 360 retries may not be enough if you want a low `polling_interval` combined with a long-running task (eg, if you want `polling_interval` to be 0.1m/6s, but running up to an hour, this requires 600 retries).

   In GitHub Enterprise environments, you may not be subject to API rate limits, so more frequent and long running checks may be less problematic from a rate limit perspective there.

Let me know if you have any questions. I'm also happy to split this into a separate PRs or remove certain pieces of the functionality if you'd prefer not to include all 3 things I tackled here. Thanks for providing this action!

wechuli commented:
Thanks for the contributions, will take a look at them shortly.

wechuli commented on src/checks/checks.ts:
```diff
@@ -75,9 +75,12 @@ export default class Checks {
             let ownCheckName = await extractOwnCheckNameFromWorkflow();
-            let gitHubActionsBotId = GitHubActionsBotId;
 
-            this.ownCheck = this.allChecks.find(check => check.name === ownCheckName && check.app.id === gitHubActionsBotId);
+            this.ownCheck = this.allChecks.find(check => check.name === ownCheckName && check.app.slug === 'github-actions');
```
Good catch, but could we export the app slug as a constant from the checksConstants.ts file in place of the App ID?

GUI replied:
Good call, this should be updated to use a constant now.

wechuli commented on src/utils/inputsExtractor.ts:
```diff
@@ -49,7 +49,7 @@ function inputsParser(): IInputs {
         parseFloat(core.getInput("polling_interval"))
     );
-    const retries: number = validateIntervalValues(parseInt(core.getInput("retries")));
+    const retries: number = parseInt(core.getInput("retries"));
 
```
I guess it makes sense to let users choose the values they want but I still want to validate that the user doesn't accidentally provide us with a negative integer value that might cause issues later on, could we instead edit the `validateIntervalValues` method to remove the maximum value and simply return the provided value if it a positive number. Some checks will need to be updated with that change.

GUI replied:
Sounds good. I've adjusted it so `validateIntervalValues` is still used here, and instead `validateIntervalValues` no longer enforces the maximum value.

GUI replied:
I just pushed up one more tweak (I forgot to update the tests around this), but I guess one question about behavior is for `Infinity`. I changed it so `Infinity` returns `1` instead of the old max value, since it seems like `Infinity` is probably a mistake, but let me know if you'd prefer to do something else for the infinite case.

wechuli replied:
Am good with that.

wechuli commented on src/checks/checks.ts:
```diff
@@ -139,6 +142,9 @@ export default class Checks {
         let inProgressQueuedWaiting = [checkStatus.IN_PROGRESS, checkStatus.QUEUED, checkStatus.WAITING]
         let anyInProgressQueuedWaiting = checks.filter(check => inProgressQueuedWaiting.includes(check.status));
         if (anyInProgressQueuedWaiting.length > 0) {
+            anyInProgressQueuedWaiting.forEach((check) => {
+                core.info(`Waiting for check completion of ${JSON.stringify(check.name)}`);
+            });
```
I am not sure about this. I believe this would log every in progress, queued or waiting check on each iteration. The logs might get noisy very quickly if you have a lot of checks/iterations

GUI replied:
Yeah, this probably stemmed more from finding this useful during debugging Enterprise, but I can see how it's probably not necessary. I've removed this part.

GUI replied:
Apologies for the flurry of comments, but I actually made one more tweak regarding logging based on using it for a bit with the logging removed: I still actually found value in having these logs to better understand how `check_include` or `check_exclude` also behaved on each iteration. So I've restored this logging, but hidden it behind a new `verbose: true` option that the action accepts (but it defaults to `false`, so the existing logging behavior should be the same). Does that seem like a decent approach? Or if you'd still like to remove this optional logging, I can also do that.

wechuli replied:
Having it as an option which defaults to false seems okay to me.

wechuli reviewed (changes requested):
Most of the changes look good, proposed a few changes. I'd like to avoid the change that logs every in progress and queued checks on every iteration though.

GUI commented:
@wechuli: Thank you for the feedback! I think I've addressed your comments, but let me know if you spot anything else or have any other questions/suggestions.

wechuli reviewed (approved):
Looks good 🚀 .

wechuli commented:
New Release pushed https://github.com/wechuli/allcheckspassed/releases/tag/v1.1.0.

Thank you for the contribution.
